author: bearinmindcat <bearincrypto@gmail.com>

diff --git a/chrome/browser/ui/android/extensions/java/src/org/chromium/chrome/browser/ui/extensions/ExtensionActionPopupContents.java b/chrome/browser/ui/android/extensions/java/src/org/chromium/chrome/browser/ui/extensions/ExtensionActionPopupContents.java
--- a/chrome/browser/ui/android/extensions/java/src/org/chromium/chrome/browser/ui/extensions/ExtensionActionPopupContents.java
+++ b/chrome/browser/ui/android/extensions/java/src/org/chromium/chrome/browser/ui/extensions/ExtensionActionPopupContents.java
@@ -117,8 +117,8 @@ public class ExtensionActionPopupContents implements Destroyable {
     }
 
     @CalledByNative
-    private boolean handleKeyboardEvent(WebContents webContents, KeyEvent event) {
-        if (mDelegate != null) {
+    private boolean handleKeyboardEvent(WebContents webContents, @Nullable KeyEvent event) {
+        if (mDelegate != null && event != null) {
             return mDelegate.handleKeyboardEvent(webContents, event);
         }
         return false;
