author: bearinmindcat <bearincrypto@gmail.com>

--- a/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/extensions/ExtensionsMenuCoordinator.java
+++ b/chrome/browser/ui/android/toolbar/java/src/org/chromium/chrome/browser/toolbar/extensions/ExtensionsMenuCoordinator.java
@@ -10,6 +10,7 @@ import android.content.res.ColorStateLis
 import android.graphics.Rect;
 import android.os.Handler;
 import android.os.Looper;
+import android.os.SystemClock;
 import android.view.LayoutInflater;
 import android.view.View;
 
@@ -98,6 +99,8 @@ public class ExtensionsMenuCoordinator
 
     @Nullable @VisibleForTesting ExtensionsMenuMediator mMediator;
 
+    private long mMenuDismissElapsedRealtimeMs;
+
     /**
      * Constructor.
      *
@@ -179,6 +182,9 @@ public class ExtensionsMenuCoordinator
         // Menu mediator is created when menu is triggered.
         mExtensionsMenuButton.setOnClickListener(
                 (view) -> {
+                    if (SystemClock.elapsedRealtime() - mMenuDismissElapsedRealtimeMs < 100) {
+                        return;
+                    }
                     TrackerFactory.getTrackerForProfile(mProfile)
                             .notifyEvent(EventConstants.EXTENSIONS_MENU_BUTTON_CLICKED);
                     createMediator();
@@ -191,6 +197,7 @@ public class ExtensionsMenuCoordinator
 
                     @Override
                     public void onPopupMenuDismissed() {
+                        mMenuDismissElapsedRealtimeMs = SystemClock.elapsedRealtime();
                         mMenuButtonPinningDelegate.requestLayoutWithViewUtils();
                         destroyMediator();
                         mExtensionModels.clear();
