author: bearinmindcat <bearincrypto@gmail.com>

--- a/components/omnibox/resources/BUILD.gn
+++ b/components/omnibox/resources/BUILD.gn
@@ -14,9 +14,6 @@ grit_strings("omnibox_pedal_synonyms") {
         "ur",
         "nb",
       ]
-  if (is_android) {
-    _locales -= [ "cy" ]
-  }
   if (is_ios) {
     _locales -= [
       "es-MX",
--- a/components/strings/BUILD.gn
+++ b/components/strings/BUILD.gn
@@ -145,9 +145,6 @@ grit_strings("components_locale_settings
                    "en-XA",
                    "ur",
                  ]
-  if (is_android) {
-    _xtb_locales -= [ "cy" ]
-  }
   if (is_ios) {
     _xtb_locales -= [
       "es-MX",
