author: bearinmindcat <bearincrypto@gmail.com>

--- a/components/search_engines/template_url_service.cc
+++ b/components/search_engines/template_url_service.cc
@@ -616,11 +616,11 @@ TemplateURLService::TemplateURLService(
     std::unique_ptr<TemplateURLServiceClient> client,
     const base::RepeatingClosure& dsp_change_callback)
     : prefs_(prefs),
-      should_autocollect_(true),
       search_engine_choice_service_(search_engine_choice_service),
       prepopulate_data_resolver_(prepopulate_data_resolver),
       regional_capabilities_service_(regional_capabilities_service),
       profile_metrics_service_(profile_metrics_service),
+      should_autocollect_(true),
       search_terms_data_(std::move(search_terms_data)),
       web_data_service_(web_data_service),
       client_(std::move(client)),
