author: bearinmindcat <bearincrypto@gmail.com>

--- a/chrome/browser/notifications/persistent_notification_handler.cc
+++ b/chrome/browser/notifications/persistent_notification_handler.cc
@@ -393,26 +393,6 @@ void PersistentNotificationHandler::OnSh
     const GURL& url,
     const std::string& notification_id,
     Profile* profile) {
-#if BUILDFLAG(IS_ANDROID)
-  safe_browsing::NotificationContentDetectionUkmUtil::
-      RecordSuspiciousNotificationInteractionUkm(
-          static_cast<int>(
-              safe_browsing::SuspiciousNotificationWarningInteractions::
-                  kShowOriginalNotification),
-          url, notification_id, profile);
-  if (base::FeatureList::IsEnabled(
-          safe_browsing::kAutoRevokeSuspiciousNotification)) {
-    auto* hcsm = HostContentSettingsMapFactory::GetForProfile(profile);
-    if (hcsm && !url.is_empty()) {
-      hcsm->SetWebsiteSettingCustomScope(
-          ContentSettingsPattern::FromURLNoWildcard(url),
-          ContentSettingsPattern::Wildcard(),
-          ContentSettingsType::SUSPICIOUS_NOTIFICATION_SHOW_ORIGINAL,
-          base::Value(base::DictValue().Set(
-              safe_browsing::kSuspiciousNotificationShowOriginalKey, true)));
-    }
-  }
-#endif
 }
 
 void PersistentNotificationHandler::OnMaybeReport(
