author: bearinmindcat <bearincrypto@gmail.com>

--- a/components/assist_ranker/ranker_url_fetcher.cc
+++ b/components/assist_ranker/ranker_url_fetcher.cc
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma clang diagnostic ignored "-Wunreachable-code"
+
 #include "components/assist_ranker/ranker_url_fetcher.h"
 
 #include <optional>
--- a/components/crash/core/app/crashpad.cc
+++ b/components/crash/core/app/crashpad.cc
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma clang diagnostic ignored "-Wunreachable-code"
+
 #include "components/crash/core/app/crashpad.h"
 
 #include <stddef.h>
--- a/components/domain_reliability/uploader.cc
+++ b/components/domain_reliability/uploader.cc
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma clang diagnostic ignored "-Wunreachable-code"
+
 #include "components/domain_reliability/uploader.h"
 
 #include <utility>
--- a/components/themes/ntp_background_service.cc
+++ b/components/themes/ntp_background_service.cc
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#pragma clang diagnostic ignored "-Wunreachable-code"
+
 #include "components/themes/ntp_background_service.h"
 
 #include <optional>
