author: bearinmindcat <bearincrypto@gmail.com>

--- a/chrome/browser/safe_browsing/android/safe_browsing_bridge.cc
+++ b/chrome/browser/safe_browsing/android/safe_browsing_bridge.cc
@@ -16,7 +16,6 @@
 #include "components/safe_browsing/content/browser/safe_browsing_service_interface.h"
 #include "components/safe_browsing/content/common/file_type_policies.h"
 #include "components/safe_browsing/core/common/hashprefix_realtime/hash_realtime_utils.h"
-#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
 #include "content/public/browser/web_contents.h"
 
 // Must come after all headers that specialize FromJniType() / ToJniType().
@@ -26,7 +25,8 @@ using base::android::JavaRef;
 
 namespace {
 
-PrefService* GetPrefService(const base::android::JavaRef<jobject>& j_profile) {
+[[maybe_unused]] PrefService* GetPrefService(
+    const base::android::JavaRef<jobject>& j_profile) {
   return Profile::FromJavaObject(j_profile)->GetPrefs();
 }
 
@@ -37,61 +37,45 @@ namespace safe_browsing {
 static int32_t JNI_SafeBrowsingBridge_UmaValueForFile(
     JNIEnv* env,
     const JavaRef<jstring>& path) {
-  base::FilePath file_path(base::android::ConvertJavaStringToUTF8(env, path));
-  return safe_browsing::FileTypePolicies::GetInstance()->UmaValueForFile(
-      file_path);
+  return 0;
 }
 
 static bool JNI_SafeBrowsingBridge_GetSafeBrowsingExtendedReportingEnabled(
     JNIEnv* env,
     const JavaRef<jobject>& j_profile) {
-  return safe_browsing::IsExtendedReportingEnabled(*GetPrefService(j_profile));
+  return false;
 }
 
 static void JNI_SafeBrowsingBridge_SetSafeBrowsingExtendedReportingEnabled(
     JNIEnv* env,
     const JavaRef<jobject>& j_profile,
-    bool enabled) {
-  safe_browsing::SetExtendedReportingPrefAndMetric(
-      GetPrefService(j_profile), enabled,
-      safe_browsing::SBER_OPTIN_SITE_ANDROID_SETTINGS);
-}
+    bool enabled) {}
 
 static bool JNI_SafeBrowsingBridge_GetSafeBrowsingExtendedReportingManaged(
     JNIEnv* env,
     const JavaRef<jobject>& j_profile) {
-  PrefService* pref_service = GetPrefService(j_profile);
-  return pref_service->IsManagedPreference(
-      prefs::kSafeBrowsingScoutReportingEnabled);
+  return false;
 }
 
 static int32_t JNI_SafeBrowsingBridge_GetSafeBrowsingState(
     JNIEnv* env,
     const JavaRef<jobject>& j_profile) {
-  return static_cast<int32_t>(
-      safe_browsing::GetSafeBrowsingState(*GetPrefService(j_profile)));
+  return 0;
 }
 
 static void JNI_SafeBrowsingBridge_SetSafeBrowsingState(
     JNIEnv* env,
     const JavaRef<jobject>& j_profile,
-    int32_t state) {
-  return safe_browsing::SetSafeBrowsingState(
-      GetPrefService(j_profile), static_cast<SafeBrowsingState>(state),
-      /*is_esb_enabled_by_account_integration=*/false);
-}
+    int32_t state) {}
 
 static void JNI_SafeBrowsingBridge_EnableSafeBrowsingSettingSetLocallyPref(
     JNIEnv* env,
-    const JavaRef<jobject>& j_profile) {
-  return safe_browsing::EnableSafeBrowsingSettingSetLocallyPref(
-      GetPrefService(j_profile));
-}
+    const JavaRef<jobject>& j_profile) {}
 
 static bool JNI_SafeBrowsingBridge_IsSafeBrowsingManaged(
     JNIEnv* env,
     const JavaRef<jobject>& j_profile) {
-  return safe_browsing::IsSafeBrowsingPolicyManaged(*GetPrefService(j_profile));
+  return false;
 }
 
 static bool JNI_SafeBrowsingBridge_IsHashRealTimeLookupEligibleInSession(
@@ -105,9 +89,6 @@ static void JNI_SafeBrowsingBridge_Repor
     content::WebContents* web_contents,
     const std::string& package_name,
     const std::string& uri) {
-  reinterpret_cast<SafeBrowsingServiceInterface*>(
-      g_browser_process->safe_browsing_service())
-      ->ReportExternalAppRedirect(web_contents, package_name, uri);
 }
 
 }  // namespace safe_browsing
