author: bearinmindcat <bearincrypto@gmail.com>

--- a/components/safe_browsing/core/browser/safe_browsing_metrics_collector.cc
+++ b/components/safe_browsing/core/browser/safe_browsing_metrics_collector.cc
@@ -24,12 +24,12 @@ using UserState = safe_browsing::SafeBro
 const int kMetricsLoggingIntervalDay = 1;
 
 // The max length of event timestamps stored in pref.
-const int kTimestampsMaxLength = 30;
+[[maybe_unused]] const int kTimestampsMaxLength = 30;
 // The quota for ESB disabled metrics. ESB disabled metrics should not be logged
 // more than the quota in a week.
 const int kEsbDisabledMetricsQuota = 3;
 // Events that are older than 30 days are removed from pref.
-const int kEventMaxDurationDay = 30;
+[[maybe_unused]] const int kEventMaxDurationDay = 30;
 // The ESB enabled duration is considered short if it's under 1 hour, long if
 // it's at least 24 hours, and medium if it's in between.
 const int kEsbShortEnabledUpperBoundHours = 1;
@@ -39,11 +39,11 @@ std::string EventTypeToPrefKey(const Eve
   return base::NumberToString(static_cast<int>(type));
 }
 
-std::string UserStateToPrefKey(const UserState& user_state) {
+[[maybe_unused]] std::string UserStateToPrefKey(const UserState& user_state) {
   return base::NumberToString(static_cast<int>(user_state));
 }
 
-base::Value TimeToPrefValue(const base::Time& time) {
+[[maybe_unused]] base::Value TimeToPrefValue(const base::Time& time) {
   return base::Int64ToValue(time.ToDeltaSinceWindowsEpoch().InSeconds());
 }
 
