author: bearinmindcat <bearincrypto@gmail.com>

---
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1021,7 +1021,6 @@ source_set("core") {
     "//chrome/browser/profiling_host",
     "//chrome/browser/renderer_host",
     "//chrome/browser/resource_coordinator",
-    "//chrome/browser/safe_browsing",
     "//chrome/browser/safe_browsing:advanced_protection",
     "//chrome/browser/search",
     "//chrome/browser/search_engines",
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -108,6 +108,10 @@ static_library("safe_browsing") {
     "//services/preferences/public/cpp",
   ]
   }
+  deps = []
+  public = []
+  sources = []
+  allow_circular_includes_from = []
 
   if (enable_extensions) {
 
