author: bearinmindcat <bearincrypto@gmail.com>

--- a/chrome/browser/download/chrome_download_manager_delegate.cc
+++ b/chrome/browser/download/chrome_download_manager_delegate.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/download/chrome_download_manager_delegate.h"
+#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
 
 #include <algorithm>
 #include <memory>
--- a/chrome/browser/download/download_target_determiner.cc
+++ b/chrome/browser/download/download_target_determiner.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/download/download_target_determiner.h"
+#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
 
 #include <optional>
 #include <string>
--- a/chrome/browser/download/download_ui_safe_browsing_util.cc
+++ b/chrome/browser/download/download_ui_safe_browsing_util.cc
@@ -3,6 +3,8 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/download/download_ui_safe_browsing_util.h"
+#include "base/files/file_path.h"
+#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
 
 #include "base/metrics/histogram_functions.h"
 #include "base/strings/stringprintf.h"
--- a/chrome/browser/file_system_access/chrome_file_system_access_permission_context.cc
+++ b/chrome/browser/file_system_access/chrome_file_system_access_permission_context.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/file_system_access/chrome_file_system_access_permission_context.h"
+#include "base/logging.h"
 
 #include <algorithm>
 #include <iterator>
--- a/chrome/browser/interstitials/enterprise_util.cc
+++ b/chrome/browser/interstitials/enterprise_util.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/interstitials/enterprise_util.h"
+#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
 
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/safe_browsing/safe_browsing_navigation_observer_manager_factory.h"
--- a/chrome/browser/profiles/profile_avatar_downloader.cc
+++ b/chrome/browser/profiles/profile_avatar_downloader.cc
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "chrome/browser/profiles/profile_avatar_downloader.h"
+#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
 
 #include <memory>
 #include <string>
