author: bearinmindcat <bearincrypto@gmail.com>

--- a/net/dns/host_resolver_manager.cc
+++ b/net/dns/host_resolver_manager.cc
@@ -151,7 +151,7 @@ const size_t kMaxHostLength = 4096;
 
 // Time between IPv6 probes, i.e. for how long results of each IPv6 probe are
 // cached.
-const int kIPv6ProbePeriodMs = 1000;
+[[maybe_unused]] const int kIPv6ProbePeriodMs = 1000;
 
 // True if |hostname| ends with either ".local" or ".local.".
 bool ResemblesMulticastDNSName(std::string_view hostname) {
@@ -174,7 +174,8 @@ bool ConfigureAsyncDnsNoFallbackFieldTri
   return kDefault;
 }
 
-base::DictValue NetLogIPv6AvailableParams(bool ipv6_available, bool cached) {
+[[maybe_unused]] base::DictValue NetLogIPv6AvailableParams(
+    bool ipv6_available, bool cached) {
   base::DictValue dict;
   dict.Set("ipv6_available", ipv6_available);
   dict.Set("cached", cached);
