author: bearinmindcat <bearincrypto@gmail.com>

--- a/components/signin/internal/identity_manager/primary_account_manager.cc
+++ b/components/signin/internal/identity_manager/primary_account_manager.cc
@@ -272,6 +272,8 @@ PrimaryAccountManager::~PrimaryAccountMa
 
 // static
 void PrimaryAccountManager::RegisterProfilePrefs(PrefRegistrySimple* registry) {
+  registry->RegisterBooleanPref("signin.allowed", true);
+  registry->RegisterStringPref("google.services.last_username", std::string());
 }
 
 // static
--- a/components/signin/public/base/signin_prefs.cc
+++ b/components/signin/public/base/signin_prefs.cc
@@ -194,6 +194,8 @@ SigninPrefs::~SigninPrefs() = default;
 void SigninPrefs::RegisterProfilePrefs(PrefRegistrySimple* registry) {
   registry->RegisterDictionaryPref(kSigninAccountPrefs);
   registry->RegisterIntegerPref(kAccountMetricsNextUnassignedId, 0);
+  registry->RegisterIntegerPref("signin.history_sync.successive_decline_count", 0);
+  registry->RegisterInt64Pref("signin.history_sync.last_declined_timestamp", 0);
 }
 
 void SigninPrefs::MigrateObsoleteSigninPrefs() {
