author: bearinmindcat <bearincrypto@gmail.com>

--- a/components/signin/internal/identity_manager/account_tracker_service.cc
+++ b/components/signin/internal/identity_manager/account_tracker_service.cc
@@ -106,7 +106,7 @@ void RemoveImage(const base::FilePath& i
   }
 }
 
-std::string AccountsToString(
+[[maybe_unused]] std::string AccountsToString(
     const std::map<CoreAccountId, AccountInfo>& accounts) {
   std::stringstream result;
   result << "[";
