description: Disable OfflineIndicator
author: Wengling Chen <feiyu2817@gmail.com>
author: bearinmindcat <bearincrypto@gmail.com>
date: Tue, 22 Jul 2026
---
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedRootUiCoordinator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabbed_mode/TabbedRootUiCoordinator.java
@@ -2049,7 +2049,7 @@ public class TabbedRootUiCoordinator ext
                     }
                 };
 
-        if (!CommandLine.getInstance()
+        if (false && !CommandLine.getInstance()
                 .hasSwitch(ContentSwitches.FORCE_ONLINE_CONNECTION_STATE_FOR_INDICATOR)) {
             mOfflineIndicatorController =
                     new OfflineIndicatorControllerV2(
