author: bearinmindcat <bearincrypto@gmail.com>

--- a/extensions/common/extension_features.cc
+++ b/extensions/common/extension_features.cc
@@ -269,7 +269,7 @@ BASE_FEATURE(kOptimizeWebRequestProxy, b
 #if BUILDFLAG(IS_ANDROID)
 BASE_FEATURE(kEnableExtensionsAndroid,
              "EnableExtensionsAndroid",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 #endif
 
 BASE_FEATURE(kUseSha256ForExtensionHashes, base::FEATURE_DISABLED_BY_DEFAULT);
